printer.go

```diff
@@ -14,28 +14,35 @@ const (
 	// DefaultIndent is the default indent string used to indent nested values.
 	DefaultIndent = "    "
 
-	// DefaultRecursionMarker is the default string to displayed when recursion is
-	// detected within a Go value.
+	// DefaultRecursionMarker is the default string to displayed when recursion
+	// is detected within a Go value.
 	DefaultRecursionMarker = "<recursion>"
 )
 
 // Filter is a function that provides custom formatting logic for specific
 // values.
 //
-// It writes a formatted representation of v to w, and returns the number of
-// bytes written.
+// It writes a formatted representation of v to w, and returns true if the
+// filter wrote any output to w.
 //
-// If the number of bytes written is non-zero, the default formatting logic is
+// If any output has been written by the filter, the default formatting logic is
 // skipped.
 //
+// format function is used whenever the filter needs to render a value with
+// default formatting.
+//
 // Particular attention should be paid to the v.IsUnexported field. If this flag
 // is true, many operations on v.Value are unavailable.
-type Filter func(w io.Writer, v Value) (int, error)
```

Comment:
I think the docs for `Filter` need to be reworded so that it no longer refers to "output being written", as that's no longer the deciding factor at all.

Perhaps it could be explained as the filter "intercepting" regular rendering by returning `true`, or something more along those lines.

Reply:
How about changing these two paragraphs in the comments to this?
```
// The filter returns true if it intercepted the value formatting. In this
// case the default formatting logic should be skipped.
```

Reply:
The comment above is outdated, right?

Reply:
This bit of comment is out of date. The rest is pretty good.
```
// Particular attention should be paid to the v.IsUnexported field. If this flag
// is true, many operations on v.Value are unavailable.
```

Reply:
This is still true atm, though I did create #31 last night to make this whole filter-writing experience better.